config/buses/natss/100-natss.yaml

```diff
@@ -0,0 +1,106 @@
+apiVersion: v1
+data:
+  gnatsd.conf: |
+    # configuration file used to override default NATS server settings
+  stan.conf: |
+    # content of configuration file used to override default NATS Streaming server settings
+kind: ConfigMap
+metadata:
+  labels:
+    app: nats-streaming
+  name: nats-streaming
+  namespace: default
+---  
+apiVersion: v1
+kind: Service
+metadata:
+  name: nats-streaming-headless
+  namespace: default
+  labels:
+    app: nats-streaming
+spec:
+#  type: NodePort
+  type: ClusterIP
+#  clusterIP: None
+  ports:
+  - name: client
+    port: 4222
+    protocol: TCP
+    targetPort: client
+  selector:
+    app: nats-streaming
+  sessionAffinity: None
+  selector:
+    app: nats-streaming
+---    
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: nats-streaming
+  namespace: default
+  labels:
+    app: nats-streaming
+spec:
+  serviceName: nats-streaming-headless
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nats-streaming
+  serviceName: nats-streaming-headless
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "true"
+      labels:
+        app: nats-streaming
+    spec:
+      containers:
+        - name: nats-streaming
+          image: nats-streaming:0.11.0
+          imagePullPolicy: IfNotPresent
+          args:
+          - -D
+          - -SD
+          - --cluster_id=knative-nats-streaming
```

Comment:
is this something that needs to be possibly updated when deploying?

Reply:
Yes, this is a kind of a shared "security context" between all NATS Streaming servers running in a cluster and all their clients. Now it's only a simple deployment without clustering.